src/components/Tour/TourBubble.tsx

```diff
@@ -0,0 +1,188 @@
+import { Flex, Link, useThemeContext } from "@radix-ui/themes";
+import { useAppSelector, useAppDispatch } from "../../hooks";
+import { RootState } from "../../app/store";
+import { close, next } from "../../features/Tour";
+import { useWindowDimensions } from "../../hooks/useWindowDimensions";
+import { TourBox } from "./TourBox";
+import { TourTitle } from "./TourTitle";
+import { ReactNode, useEffect, useState } from "react";
+
+export type TourBubbleProps = {
+  text: string;
+  step: number;
+  down: boolean;
+  isPointing?: boolean;
+  target: HTMLElement | null;
+  containerWidth?: string;
+  onPage: string;
+  page: string;
+  deltaY?: number;
+  children?: ReactNode;
+  onNext?: () => void;
+};
+
+export function TourBubble({
+  text,
+  step,
+  target,
+  down,
+  containerWidth,
+  onPage,
+  page,
+  isPointing,
+  deltaY,
+  children,
+  onNext,
+}: TourBubbleProps) {
+  const dispatch = useAppDispatch();
+  const state = useAppSelector((state: RootState) => state.tour);
+  const { width: windowWidth, height: windowHeight } = useWindowDimensions();
+  const [pos, setPos] = useState<DOMRect | undefined>(undefined);
+  const appearance = useThemeContext().appearance;
+
+  const isBubbleOpen = state.type === "in_progress" && state.step === step;
+
+  if (isPointing === undefined) {
+    isPointing = true;
+  }
+
+  // TODO: find a better way of doing this
```

Comment:
The last time i did a tour, we used the popover and hover card components with condition rendering.